src/proxy/Map.ts

```diff
@@ -56,4 +124,269 @@ export class Map<K, V> extends BaseProxy implements IMap<K, V> {
     isEmpty(): Q.Promise<boolean> {
         return this.encodeInvokeOnRandomTarget<boolean>(MapIsEmptyCodec);
     }
+
+    getAll(keys: K[]): Q.Promise<any[]> {
+        var partitionService = this.client.getPartitionService();
+        var partitionsToKeys: {[id: string]: any} = {};
+        var key: K;
```

Comment:
There is similar logic of partition ID mapping in putAll, can you extract into separate method?